Sources/NIOCore/ByteBuffer-binaryEncodedLengthPrefix.swift

```diff
@@ -0,0 +1,337 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+#if os(Windows)
+import ucrt
```

Comment:
@hamzahrmalik why do we need the libcs here?

Reply:
for the memmove function

Reply:
Okay, in this case we should remove the imports here.

- we shouldn't use `*Unsafe*` here, the most minimal subset of `ByteBuffer-core.swift` needs to provide enough
- `memmove` isn't necessary to be called directly. `UnsafeRawBufferPointer` can move memory with a higher-level API (I think it's `bufferPointer[0..<100] = bufferPointer[2..<102]` or something like that; but again, no Unsafe in here please)